File: src/main/java/com/dingdongdeng/coinautotrading/domain/repository/ExchangeKeyRepository.java
package com.dingdongdeng.coinautotrading.domain.repository;

import com.dingdongdeng.coinautotrading.domain.entity.ExchangeKey;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ExchangeKeyRepository extends JpaRepository<ExchangeKey, Long> {

  List<ExchangeKey> findByPairId(String pairId);

  List<ExchangeKey> findByUserId(String userId);
}

File: app-service/src/main/kotlin/com/dingdongdeng/autotrading/domain/exchange/repository/ExchangeKeyRepository.kt
package com.dingdongdeng.autotrading.domain.exchange.repository

import com.dingdongdeng.autotrading.domain.exchange.entity.ExchangeKey
import com.dingdongdeng.autotrading.infra.common.type.ExchangeType
import org.springframework.data.jpa.repository.JpaRepository

interface ExchangeKeyRepository : JpaRepository<ExchangeKey, Long> {

    fun findByKeyPairId(keyPairId: String): List<ExchangeKey>

    fun findAllByExchangeTypeAndUserId(exchangeType: ExchangeType, userId: Long): List<ExchangeKey>
}